File: cpp/leetcode/083.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* ptr = head;
        while (ptr) {
            while(ptr->next && ptr->val == ptr->next->val) 
                ptr->next = ptr->next->next;
            ptr = ptr->next;
        }
        return head;
    }
};

File: python/leetcode/083.py
class Solution:
    def deleteDuplicates(self, head):
        ptr = head
        while ptr:
            while ptr.next and ptr.next.val == ptr.val:
                ptr.next = ptr.next.next     # skip duplicated node
            ptr = ptr.next   # not duplicate of ptrrent node, move to next node
        return head
